qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -158,29 +164,68 @@ def __init__(
 
         self._quantum_instance = None
         if quantum_instance is not None:
+            warnings.warn(
+                "The quantum_instance argument has been superseded by the sampler argument. "
+                "This argument will be deprecated in a future release and subsequently "
+                "removed after that.",
+                category=PendingDeprecationWarning,
+                stacklevel=2,
+            )
             self.quantum_instance = quantum_instance
 
+        self._sampler = sampler
+
         self._sample_from_iterations = sample_from_iterations
         self._iterations_arg = iterations
 
     @property
+    @deprecate_function(
+        "The Grover.quantum_instance getter is pending deprecation. "
+        "This property will be deprecated in a future release and subsequently "
+        "removed after that.",
+        category=PendingDeprecationWarning,
+    )
     def quantum_instance(self) -> Optional[QuantumInstance]:
-        """Get the quantum instance.
+        """Pending deprecation: Get the quantum instance.
         Returns:
             The quantum instance used to run this algorithm.
         """
         return self._quantum_instance
 
     @quantum_instance.setter
+    @deprecate_function(
+        "The Grover.quantum_instance setter is pending deprecation. "
+        "This property will be deprecated in a future release and subsequently "
+        "removed after that.",
+        category=PendingDeprecationWarning,
+    )
     def quantum_instance(self, quantum_instance: Union[QuantumInstance, Backend]) -> None:
-        """Set quantum instance.
+        """Pending deprecation: Set quantum instance.
         Args:
             quantum_instance: The quantum instance used to run this algorithm.
         """
         if isinstance(quantum_instance, Backend):
             quantum_instance = QuantumInstance(quantum_instance)
         self._quantum_instance = quantum_instance
 
+    @property
+    def sampler(self) -> Optional[BaseSampler]:
+        """Get the sampler.
+
+        Returns:
+            The sampler used to run this algorithm.
+        """
+        return self._sampler
+
+    @sampler.setter
+    def sampler(self, sampler: BaseSampler) -> None:
```

Comment:
Technically, setting it to `None` is also supported
```suggestion
    def sampler(self, sampler: Optional[BaseSampler]) -> None:
```